jerry-core/CMakeLists.txt

```diff
@@ -25,220 +52,174 @@ project (JerryCore C ASM)
+      CONFIG_ECMA_COMPACT_PROFILE_DISABLE_ERROR_BUILTINS
+      CONFIG_ECMA_COMPACT_PROFILE_DISABLE_ARRAY_BUILTIN
+      CONFIG_ECMA_COMPACT_PROFILE_DISABLE_MATH_BUILTIN
+      CONFIG_ECMA_COMPACT_PROFILE_DISABLE_JSON_BUILTIN)
+
+else()
+  message(FATAL_ERROR "FEATURE_PROFILE='${FEATURE_PROFILE}' doesn't supported")
+endif()
 
 # Include directories
- set(INCLUDE_CORE
-     ${CMAKE_SOURCE_DIR}/jerry-core
-     ${CMAKE_SOURCE_DIR}/jerry-core/lit
-     ${CMAKE_SOURCE_DIR}/jerry-core/rcs
-     ${CMAKE_SOURCE_DIR}/jerry-core/jmem
-     ${CMAKE_SOURCE_DIR}/jerry-core/vm
-     ${CMAKE_SOURCE_DIR}/jerry-core/ecma/builtin-objects
-     ${CMAKE_SOURCE_DIR}/jerry-core/ecma/base
-     ${CMAKE_SOURCE_DIR}/jerry-core/ecma/operations
-     ${CMAKE_SOURCE_DIR}/jerry-core/parser/js
-     ${CMAKE_SOURCE_DIR}/jerry-core/parser/regexp
-     ${CMAKE_SOURCE_DIR}/jerry-core/jrt)
-
- # Third-party
-  # Valgrind
-   set(INCLUDE_THIRD_PARTY_VALGRIND ${CMAKE_SOURCE_DIR}/third-party/valgrind)
+set(INCLUDE_CORE
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_SOURCE_DIR}/lit
+    ${CMAKE_CURRENT_SOURCE_DIR}/rcs
+    ${CMAKE_CURRENT_SOURCE_DIR}/jmem
+    ${CMAKE_CURRENT_SOURCE_DIR}/vm
+    ${CMAKE_CURRENT_SOURCE_DIR}/ecma/builtin-objects
+    ${CMAKE_CURRENT_SOURCE_DIR}/ecma/base
+    ${CMAKE_CURRENT_SOURCE_DIR}/ecma/operations
+    ${CMAKE_CURRENT_SOURCE_DIR}/parser/js
+    ${CMAKE_CURRENT_SOURCE_DIR}/parser/regexp
+    ${CMAKE_CURRENT_SOURCE_DIR}/jrt)
 
 # Sources
- # Jerry core
-  file(GLOB SOURCE_CORE_API                   *.c)
-  file(GLOB SOURCE_CORE_LIT                   lit/*.c)
-  file(GLOB SOURCE_CORE_RCS                   rcs/*.c)
-  file(GLOB SOURCE_CORE_MEM                   jmem/*.c)
-  file(GLOB SOURCE_CORE_VM                    vm/*.c)
-  file(GLOB SOURCE_CORE_ECMA_BUILTINS         ecma/builtin-objects/*.c)
-  file(GLOB SOURCE_CORE_ECMA_BASE             ecma/base/*.c)
-  file(GLOB SOURCE_CORE_ECMA_OPERATIONS       ecma/operations/*.c)
-  file(GLOB SOURCE_CORE_PARSER_JS             parser/js/*.c)
-  file(GLOB SOURCE_CORE_PARSER_REGEXP         parser/regexp/*.c)
-  file(GLOB SOURCE_CORE_JRT                   jrt/*.c)
-
-  set(SOURCE_CORE_FILES
-      ${SOURCE_CORE_API}
-      ${SOURCE_CORE_LIT}
-      ${SOURCE_CORE_RCS}
-      ${SOURCE_CORE_MEM}
-      ${SOURCE_CORE_VM}
-      ${SOURCE_CORE_ECMA_BUILTINS}
-      ${SOURCE_CORE_ECMA_BASE}
-      ${SOURCE_CORE_ECMA_OPERATIONS}
-      ${SOURCE_CORE_PARSER_JS}
-      ${SOURCE_CORE_PARSER_REGEXP}
-      ${SOURCE_CORE_JRT})
-
- # Jerry port
- if(NOT ("${PORT_DIR}" STREQUAL ""))
-  file(GLOB SOURCE_PORT_FILES ${PORT_DIR}/*.c)
-  if (NOT USE_JERRY_LIBC)
-   set (DEFINES_JERRY ${DEFINES_JERRY} _BSD_SOURCE)
-  endif()
- endif()
-
- # All-in-one build
- if("${ENABLE_ALL_IN_ONE}" STREQUAL "ON")
+# Jerry core
+file(GLOB SOURCE_CORE_API                   *.c)
+file(GLOB SOURCE_CORE_LIT                   lit/*.c)
+file(GLOB SOURCE_CORE_RCS                   rcs/*.c)
```

Comment:
The rcs subdirectory does not exists anymore. Please check these and order them in alphabetic order.